src/request_queue.js

```diff
@@ -460,13 +478,17 @@ export class RequestQueue {
      *
      * @ignore
      */
-    _ensureHeadIsNonEmpty(checkModifiedAt = false, limit = Math.max(this.inProgressCount * QUERY_HEAD_BUFFER, QUERY_HEAD_MIN_LENGTH), iteration = 0) {
+    async _ensureHeadIsNonEmpty(
```

Comment:
Please add a comment about the return value

Reply:
Plus comments of params too, this function is tricky enough to deserve explanation